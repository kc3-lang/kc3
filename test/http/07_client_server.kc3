quote (server = Socket.listen("localhost", "57000")) ; void
(server = Socket.listen("localhost", "57000")) ; void
quote (client = Socket.Buf.connect("localhost", "57000")) ; void
(client = Socket.Buf.connect("localhost", "57000")) ; void
quote (server_client = Socket.Buf.accept(server)) ; void
(server_client = Socket.Buf.accept(server)) ; void
quote client_req = %HTTP.Request{method: GET,
                                 url: "/",
                                 protocol: "HTTP/1.1",
                                 headers: []}
client_req = %HTTP.Request{method: GET,
                           url: "/",
                           protocol: "HTTP/1.1",
                           headers: []}
quote HTTP.Request.buf_write(client_req, client.buf_rw.w)
HTTP.Request.buf_write(client_req, client.buf_rw.w)
quote req = HTTP.Request.buf_parse(server_client.buf_rw.r)
req = HTTP.Request.buf_parse(server_client.buf_rw.r)
quote puts(req)
puts(req)
quote res = HTTP.Response.buf_write(%HTTP.Response{body: "Hello !"}, server_client.buf_rw.w, true)
res = HTTP.Response.buf_write(%HTTP.Response{body: "Hello !"}, server_client.buf_rw.w, true)
quote client_response = HTTP.Response.buf_parse(client.buf_rw.r, true)
client_response = HTTP.Response.buf_parse(client.buf_rw.r, true)
quote puts(client_response)
puts(client_response)
quote Socket.Buf.close(server_client)
Socket.Buf.close(server_client)
quote Socket.Buf.close(client)
Socket.Buf.close(client)
quote Socket.close(server)
Socket.close(server)
