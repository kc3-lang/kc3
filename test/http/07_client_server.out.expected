(server = Socket.listen("localhost", "57000")) ; void
void
(client = Socket.Buf.connect("localhost", "57000")) ; void
void
(server_client = Socket.Buf.accept(server)) ; void
void
client_req = %HTTP.Request{method: GET}
%HTTP.Request{method: GET,
              body: ?,
              url: "/",
              protocol: "HTTP/1.1",
              client_addr: "",
              headers: [],
              cookies: [],
              params: %{}}
HTTP.Request.buf_write(client_req, client.buf_rw.w)
(Sw) 18
req = HTTP.Request.buf_parse(server_client.buf_rw.r)
%HTTP.Request{method: GET,
              body: void,
              url: "/",
              protocol: "HTTP/1.1",
              client_addr: "",
              headers: [],
              cookies: [],
              params: %{}}
puts(req)
%HTTP.Request{method: GET,
              body: void,
              url: "/",
              protocol: "HTTP/1.1",
              client_addr: "",
              headers: [],
              cookies: [],
              params: %{}}
(Sw) 225
res = HTTP.Response.buf_write(%HTTP.Response{body: "Hello !"}, server_client.buf_rw.w, true)
(Sw) 70
client_response = HTTP.Response.buf_parse(client.buf_rw.r, true)
%HTTP.Response{protocol: "HTTP/1.1",
               code: (U16) 200,
               message: "OK",
               headers: [{"Content-Type", "text/html"},
                         {"Content-Length", "7"}],
               body: "Hello !"}
puts(client_response)
%HTTP.Response{protocol: "HTTP/1.1",
               code: (U16) 200,
               message: "OK",
               headers: [{"Content-Type", "text/html"},
                         {"Content-Length", "7"}],
               body: "Hello !"}
(Sw) 238
Socket.Buf.close(server_client)
void
Socket.Buf.close(client)
void
Socket.close(server)
void
